interview-questions.md

```diff
@@ -7,53 +7,64 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a block in Ruby?
 
-  Your answer:
+  Your answer: 
+    A ruby block is a method that allows you to iterate through an parameter which will allow you to extract information or manipulate data. Ruby blocks can take 3 arguments value, index and array which can be manipulate.
 
   Researched answer:
+     A ruby an anonymous block is a method that allows you to iterate through an parameter which will allow you to extract information or manipulate data. Ruby blocks can take 3 arguments value, index and array which can be manipulate.  Blocks use the dot notation followed by a do/end.
 
 
 
 2. What is a gem?
 
   Your answer:
+    It is my understanding that a GEM is similar to YARN.  A GEM brings in code snip its or other library to enhance Ruby.
 
   Researched answer:
-
+    Ruby GEM's are software package handler similar to YARN.  It allows the user to download libraries that are open source.
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer:  
+    Ruby on Rails is a Ruby framework that give you all the tools to create an app and configure it to a rails data base. It is often used by start ups to bring up web applications online quickly.
 
-  Researched answer:
 
+  Researched answer:
+    Ruby on Rails is a full stack framework that give you all the tools to create an app and configure it to a rails data base. It is often used by start ups to bring up web applications online quickly.  Ruby on Rails also uses the model view controller pattern.
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer:  
+    I belive that a realational database is where data is stored a rows and colums.  Also in order to interact with this data you must use key words in order to access the data and filiter/sort trough possibley hundreds of lines of data to make sure that you are getting the correct data.  There other types of databases, I am familiar with postgresql and sql
 
   Researched answer:
+    Realational database are very similar to excel spreadsheet where data is stored in columsn and rows.  Every set of columns and rows are considered a table.  You can imput all types of data into a database, such as strings and numbers.  Each row of data is uniquly identified by a key that can not be reasigned.  There are multiple types of database however most popular are Oracle and MySQL.
```

Comment:
Nice job. The other general category of databases is called a non-relational or NOSQL database. It has a different way of organizing content.